### PR TITLE
Implement sort list popup

### PR DESCRIPTION
**Pull request by lucyxubroad**
# Summary

Implement sort list popup screen
# Addressed Issues

#12 
#13 

# Demonstration

ezgif com-video-to-gif-9

### CODE REVIEW

File: Flick/Cells/SortOptionTableViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+//  Flick
+//
+//  Created by Lucy Xu on 6/20/20.
+//  Copyright © 2020 flick. All rights reserved.
+//
+
+import UIKit
+
+protocol SortOptionDelegate: class {
+    func setSortSelection(at index: Int, for sortDirection: SortDirection )
```
Comment (haiyingweng) [inline Flick/Cells/SortOptionTableViewCell.swift:12]: extra space after `SortDirection `

File: Flick/Cells/SortOptionTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    private func setupConstraints() {
+        let buttonHeight = 9.5
+
+        sortLabel.snp.makeConstraints { make in
+            make.centerY.equalToSuperview()
+        }
+
+        descendButton.snp.makeConstraints { make in
+            make.trailing.equalToSuperview()
+            make.centerY.equalToSuperview()
```
Comment (haiyingweng) [inline Flick/Cells/SortOptionTableViewCell.swift:75]: make.centerY.trailing.equalToSuperview()

File: Flick/Cells/SortOptionTableViewCell.swift [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        sortLabel.textColor = sortSelection.sortDirection == .unselected ? .darkBlueGray2 : .gradientPurple
+        if (sortSelection.sortDirection == .ascending) {
+            ascendButton.setImage(UIImage(named: "filledUpArrow"), for: .normal)
+            descendButton.setImage(UIImage(named: "downArrow"), for: .normal)
+        } else if (sortSelection.sortDirection == .descending) {
+            ascendButton.setImage(UIImage(named: "upArrow"), for: .normal)
+            descendButton.setImage(UIImage(named: "filledDownArrow"), for: .normal)
+        } else {
+            ascendButton.setImage(UIImage(named: "upArrow"), for: .normal)
+            descendButton.setImage(UIImage(named: "downArrow"), for: .normal)
```
Comment (haiyingweng) [inline Flick/Cells/SortOptionTableViewCell.swift:58]: could we simplify into like 
```swift
let ascentImageName = sortSelection.sortDirection == .ascending ? "filledUpArrow" : "upArrow"
let descendImageName = sortSelection.sortDirection == .descending ? "filledDownArrow" : "downArrow"
ascendButton.setImage(UIImage(named: ascentImageName), for: .normal)
descendButton.setImage(UIImage(named: descendImageName), for: .normal)
```

File: Flick/Controllers/ListViewController.swift [hunk 4]
```diff
@@ -193,3 +195,26 @@ extension ListViewController: UICollectionViewDelegateFlowLayout {
...
+    }
+
+    func editMedia() {
+        print("edit media")
+    }
+
+    func sortMedia() {
+        sortListModalView = SortListModalView()
+        sortListModalView.delegate = self
+        navigationController?.navigationBar.layer.zPosition = -1;
```
Comment (haiyingweng) [inline Flick/Controllers/ListViewController.swift:211]: what does this do?

File: Flick/Views/SortListModalView.swift [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+        sortOptionsTableView.register(SortOptionTableViewCell.self, forCellReuseIdentifier: sortOptionCellReuseIdentifier)
+        sortOptionsTableView.separatorStyle = .none
+        containerView.addSubview(sortOptionsTableView)
+
+        containerView.backgroundColor = .white
+        containerView.layer.cornerRadius = 24
+        addSubview(containerView)
+
+        setupViews()
+
```
Comment (haiyingweng) [inline Flick/Views/SortListModalView.swift:67]: extra newline

File: Flick/Views/SortListModalView.swift [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+            self.containerView.transform = .identity
+        })
+
+    }
+
+    @objc func dismiss() {
+        UIView.animate(withDuration: 0.15, animations: {
+            self.containerView.alpha = 0
+            self.containerView.transform = CGAffineTransform(scaleX: 0.8, y: 0.8)
+            self.backgroundColor = UIColor(red: 63/255, green: 58/255, blue: 88/255, alpha: 0)
```
Comment (haiyingweng) [inline Flick/Views/SortListModalView.swift:116]: if alpha is 0, is it just clear?

File: Flick/Controllers/ListViewController.swift (hunk 4 above)
Comment (lucyxubroad) [reply Flick/Controllers/ListViewController.swift:211]: Hm, this is actually not working 100% the way I want. but right now, when I show the popup, the the navigation bar does not get covered. so setting z index moves it to the back so it gets covered. but right now it ends up getting covered by the main view too.. :/

File: Flick/Views/SortListModalView.swift (hunk 6 above)
Comment (lucyxubroad) [reply Flick/Views/SortListModalView.swift:116]: its for the transition effect. so we gradually fade out the background overlay

File: Flick/Cells/SortOptionTableViewCell.swift (hunk 3 above)
Comment (lucyxubroad) [reply Flick/Cells/SortOptionTableViewCell.swift:58]: LOL yes. i think I had something like this at one point (y)

**Pull request merged**